app/src/main/kotlin/com/wire/android/migration/userDatabase/ScalaMessageDAO.kt

```diff
@@ -102,49 +103,39 @@ class ScalaMessageDAO(
-                    "$ASSETS_TABLE_NAME.$COLUMN_NAME AS $ASSET_ALIAS_PREFIX$COLUMN_NAME, " +
-                    "$ASSETS_TABLE_NAME.$COLUMN_SIZE AS $ASSET_ALIAS_PREFIX$COLUMN_SIZE " +
-                    "FROM $MESSAGES_TABLE_NAME " +
-                    "LEFT JOIN $ASSETS_TABLE_NAME ON $MESSAGES_TABLE_NAME.$COLUMN_ASSET_ID = $ASSETS_TABLE_NAME.$COLUMN_ID " +
-                    "WHERE $MESSAGES_TABLE_NAME.$COLUMN_CONVERSATION_ID = ?" +
-                    "ORDER BY $MESSAGES_TABLE_NAME.$COLUMN_TIME ASC ", arrayOf(scalaConversation.id)
+            "SELECT *" + // Assets are required, otherwise we get exception "requesting column name with table name".
+                    "FROM $MESSAGES_TABLE_NAME AS m " +
+                    "LEFT JOIN $ASSETS_TABLE_NAME ON m.$COLUMN_ASSET_ID = $ASSETS_TABLE_NAME.$COLUMN_ID " +
+                    "WHERE m.$COLUMN_CONVERSATION_ID = ?" +
+                    "ORDER BY m.$COLUMN_TIME ASC ", arrayOf(scalaConversation.id)
         )
         return try {
             if (!cursor.moveToFirst()) {
                 emptyList()
             } else {
-                val idIndex = cursor.getColumnIndex("$MESSAGE_ALIAS_PREFIX$COLUMN_ID")
-                val timeIndex = cursor.getColumnIndex("$MESSAGE_ALIAS_PREFIX$COLUMN_TIME")
-                val editTimeIndex = cursor.getColumnIndex("$MESSAGE_ALIAS_PREFIX$COLUMN_EDIT_TIME").orNullIfNegative()
-                val userIdIndex = cursor.getColumnIndex("$MESSAGE_ALIAS_PREFIX$COLUMN_USER_ID")
-                val clientIdIndex = cursor.getColumnIndex("$MESSAGE_ALIAS_PREFIX$COLUMN_CLIENT_ID")
-                val contentIndex = cursor.getColumnIndex("$MESSAGE_ALIAS_PREFIX$COLUMN_CONTENT")
-                val protoIndex = cursor.getColumnIndex("$MESSAGE_ALIAS_PREFIX$COLUMN_PROTO_BLOB")
-                val assetNameIndex = cursor.getColumnIndex("$ASSET_ALIAS_PREFIX$COLUMN_NAME")
-                val assetSizeIndex = cursor.getColumnIndex("$ASSET_ALIAS_PREFIX$COLUMN_SIZE")
+                val timeIndex = cursor.getColumnIndex("$COLUMN_TIME")
+                val editTimeIndex = cursor.getColumnIndex("$COLUMN_EDIT_TIME").orNullIfNegative()
+                val userIdIndex = cursor.getColumnIndex("$COLUMN_USER_ID")
+                val clientIdIndex = cursor.getColumnIndex("$COLUMN_CLIENT_ID").orNullIfNegative()
+                val contentIndex = cursor.getColumnIndex("$COLUMN_CONTENT").orNullIfNegative()
+                val protoIndex = cursor.getColumnIndex("$COLUMN_PROTO_BLOB").orNullIfNegative()
+                val assetNameIndex = cursor.getColumnIndex("$COLUMN_NAME").orNullIfNegative()
+                val assetSizeIndex = cursor.getColumnIndex("$COLUMN_SIZE").orNullIfNegative()
                 val accumulator = mutableListOf<ScalaMessageData>()
                 do {
                     accumulator += ScalaMessageData(
-                        id = cursor.getStringOrNull(idIndex).orEmpty(),
+                        id = cursor.getString(0),
```

Comment:
are you sure about accessing the index directly without checking?

Reply:
For ids, it is fine since it is always the first column in the table, other columns' position may change, but there is always 0, and i had to hard-code this way because when we LEFT JOIN messages with asset tables we get 2 columns that are called the same "_id"

Reply:
I think aliases can help in the case when there are multiple columns with the same name, but this is also fine I guess